### PR TITLE
Spanner: Added extra documentation to the DatabaseClient.singleUse methods

### LINKED ISSUE #4212: Spanner: Better document the difference between singleUse() and singleUseReadOnlyTransaction().
Hi,
I think the documentation of the difference between `DatabaseClient.singleUse()` and `DatabaseClient.singleUseReadOnlyTransaction()` needs to be more verbose.

This is the documentation for both of these:
> Returns a context in which a single read can be performed using TimestampBound.strong() concurrency.

> Returns a read-only transaction context in which a single read or query can be performed using TimestampBound.strong() concurrency.

By looking at the code, the only difference I can really find is that `ReadOnlyTransaction` (returned by `singleUseReadOnlyTransaction()`) holds the `read_timestamp` selected internally by Spanner once the `Read` or query finishes. If I'm reading it correctly, that means that `singleUse()` is not less transactional than `singleUseReadOnlyTransaction()` is. Is there any reason (beyond compatibility) not to remove `singleUse()` from the `DatabaseClient`?

### PR DESCRIPTION
**Pull request by olavloite**
Fixes #4212

### CODE REVIEW

File: google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java [hunk 1]
```diff
@@ -85,34 +85,43 @@
...
   /**
    * Returns a context in which a single read can be performed using {@link TimestampBound#strong()}
-   * concurrency.
+   * concurrency. This method will return a {@link ReadContext} that will not return the read
+   * timestamp that was used by Cloud Spanner. If you want to be able to access the read timestamp,
+   * you should use the method {@link #singleUseReadOnlyTransaction()}.
    *
    * <p>Example of single use.
    *
-   * <pre>{@code
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java]: Do we need to remove `@code`?

Comment (olavloite) [reply google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java]: It's no removed, but changed by the formatter into '&#64;code'.

Comment (kolea2) [reply google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java]: That's odd - I tried running it myself and didn't see the same output (stayed the same). Can you try resubmitting as `@code` and see if the linter passes?

Comment (olavloite) [reply google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseClient.java]: The wrong formatting was first introduced by my IDE (Eclipse), and the mvn plugin did not automatically change it back to the original value. It's now changed back to `@code`.

**Pull request merged**